src/search.ts

```diff
@@ -36,6 +36,13 @@ export interface SearchResult {
     lastResultUrl?: string;
 }
 
+export interface SearchFilter {
+    key: string;
+    value: string[];
+    operator: '==' | '!=' | '>' | '<' | '>=' | '<=';
+    logicalOperator?: 'AND' | 'OR';

```

Comment:
`logicalOperator` wasn't originally in the tech design. I added it to allow us to filter by the Patient's full url **or** the Patient's relative url.

This is reference code for `getSearchFilterBasedOnIdentity`
```
   async getSearchFilterBasedOnIdentity(request: GetSearchFilterBasedOnIdentity): Promise<SearchFilter[]> {
        const fhirUser = this.getFhirUser(request.userIdentity);
        const { hostname, resourceType, id } = fhirUser;

        if (resourceType !== 'Practitioner' && ['search-type', 'search-system'].includes(request.operation)) {
            return [
                {
                    key: '_reference',
                    value: [`{resourceType}/${id}`, `${hostname}${resourceType}/${id}`],
                    operator: '==',
                    logicalOperator: 'OR',
                },
            ];
        }

        return [];
    }
```